### ISSUE TITLE
Cockatrice 2.9.0 crash when we enable sound on MacOS macOS 14.5 (stack trace available)

### ISSUE THREAD

**Issue by alex-mehdi-dev** [Cockatrice/Cockatrice#5078]
```
Client Version: 2.9.0 (2023-09-14)
Client Operating System: macOS 14.5
Build Architecture: 64-bit
Qt Version: 6.5.1
System Locale: en_FR
Install Mode: Standard
---------------------------------------------------------------------------
Theme changed: "Mine"
SoundEngine: disabling sound
Sound theme changed: "Default"
SoundEngine: disabling sound
Unable to load qt translation "qt_" at "/Applications/cockatrice.app/Contents/Resources/translations"
Unable to load "cockatrice" translation "cockatrice_" at "/Applications/cockatrice.app/Contents/MacOS/../Resources/translations"
main(): starting main program
main(): MainWindow constructor finished
Spoilers Disabled
main(): ui.show() finished
PictureLoader: cache fail for "_trice_card_back_401499"
Startup: found config with current version
CardDatabase::loadCardDatabases start
[CockatriceXml4Parser] Trying to parse:  "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/cards.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/cards.xml" Status = 0 Cards = 29709 Sets = 760 "903ms"
[CockatriceXml4Parser] Trying to parse:  "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/tokens.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/tokens.xml" Status = 0 Cards = 30542 Sets = 760 "23ms"
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/spoiler.xml" Status = 4 Cards = 30542 Sets = 760 "0ms"
Loading Custom Set 0 ( "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/customsets/TK.xml" )
[CockatriceXml4Parser] Trying to parse:  "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/customsets/TK.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/customsets/TK.xml" Status = 0 Cards = 30542 Sets = 760 "0ms"
CardDatabase::loadCardDatabases success
```
_______________________________________________________________________________________

The application crash when we enable sound. Here is an extract of the stack trace:

```
Process:               cockatrice [30806]
Path:                  /Applications/cockatrice.app/Contents/MacOS/cockatrice
Identifier:            com.cockatrice.Cockatrice
Version:               2.9.0 (2.9.0)
Code Type:             X86-64 (Translated)
OS Version:            macOS 14.5 (23F79)

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   ???                           	    0x7ff89f352a78 ???
1   libsystem_kernel.dylib        	    0x7ff80f07714a __pthread_kill + 10
2   libsystem_pthread.dylib       	    0x7ff80f0afebd pthread_kill + 262
3   libsystem_c.dylib             	    0x7ff80efd5a79 abort + 126
4   QtCore                        	       0x111d69707 qAbort() + 9
5   QtCore                        	       0x111d6c677 qt_message_fatal(QtMsgType, QMessageLogContext const&, QString const&) + 9
6   QtCore                        	       0x111fe1b79 QMessageLogger::fatal(char const*, ...) const + 213
7   QtMultimedia                  	       0x10e760aab QPlatformMediaIntegration::instance() + 1397
8   QtMultimedia                  	       0x10e762679 QMediaPlayer::QMediaPlayer(QObject*) + 77
9   cockatrice                    	       0x104c1c48a SoundEngine::soundEnabledChanged() + 554
10  QtCore                        	       0x111df175d void doActivate<false>(QObject*, int, void**) + 1317
11  cockatrice                    	       0x104bebf1b SettingsCache::setSoundEnabled(int) + 107
12  QtCore                        	       0x111df175d void doActivate<false>(QObject*, int, void**) + 1317
13  QtWidgets                     	       0x10f2e9161 QCheckBox::checkStateSet() + 143
14  QtWidgets                     	       0x10f2d8709 QAbstractButtonPrivate::click() + 137
15  QtWidgets                     	       0x10f2d9388 QAbstractButton::mouseReleaseEvent(QMouseEvent*) + 198
16  QtWidgets                     	       0x10f22a9f0 QWidget::event(QEvent*) + 1434
17  QtWidgets                     	       0x10f1e9489 QApplicationPrivate::notify_helper(QObject*, QEvent*) + 247
18  QtWidgets                     	       0x10f1eb7e3 QApplication::notify(QObject*, QEvent*) + 6049
19  QtCore                        	       0x111db6a09 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 157
20  QtWidgets                     	       0x10f1e99e3 QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) + 897
21  QtWidgets                     	       0x10f23d0eb QWidgetWindow::handleMouseEvent(QMouseEvent*) + 2011
22  QtWidgets                     	       0x10f23c0d0 QWidgetWindow::event(QEvent*) + 172
23  QtWidgets                     	       0x10f1e9489 QApplicationPrivate::notify_helper(QObject*, QEvent*) + 247
24  QtWidgets                     	       0x10f1ea256 QApplication::notify(QObject*, QEvent*) + 532
25  QtCore                        	       0x111db6a09 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 157
26  QtGui                         	       0x110d812b3 QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) + 1763
27  QtGui                         	       0x110dc9540 QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 110
28  libqcocoa.dylib               	       0x10e2a8d03 QCocoaEventDispatcherPrivate::postedEventsSourceCallback(void*) + 423
29  CoreFoundation                	    0x7ff80f18d7d6 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
30  CoreFoundation                	    0x7ff80f18d779 __CFRunLoopDoSource0 + 157
31  CoreFoundation                	    0x7ff80f18d548 __CFRunLoopDoSources0 + 215
32  CoreFoundation                	    0x7ff80f18c1b8 __CFRunLoopRun + 919
33  CoreFoundation                	    0x7ff80f18b859 CFRunLoopRunSpecific + 557
34  HIToolbox                     	    0x7ff81a07ea09 RunCurrentEventLoopInMode + 292
35  HIToolbox                     	    0x7ff81a07e646 ReceiveNextEventCommon + 201
36  HIToolbox                     	    0x7ff81a07e561 _BlockUntilNextEventMatchingListInModeWithFilter + 66
37  AppKit                        	    0x7ff81275dc61 _DPSNextEvent + 880
38  AppKit                        	    0x7ff813071dc0 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1273
39  AppKit                        	    0x7ff8129f0e6a -[NSApplication _doModalLoop:peek:] + 297
40  AppKit                        	    0x7ff812c2c3aa __33-[NSApplication runModalSession:]_block_invoke_2 + 69
41  AppKit                        	    0x7ff812c2c352 __33-[NSApplication runModalSession:]_block_invoke + 78
42  AppKit                        	    0x7ff8129ef245 _NSTryRunModal + 100
43  AppKit                        	    0x7ff812c2c24f -[NSApplication runModalSession:] + 132
44  libqcocoa.dylib               	       0x10e2a75c6 QCocoaEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 618
45  QtCore                        	       0x111dbd508 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 362
46  QtWidgets                     	       0x10f3d6e50 QDialog::exec() + 274
47  cockatrice                    	       0x104c76580 MainWindow::actSettings() + 32
48  QtCore                        	       0x111df175d void doActivate<false>(QObject*, int, void**) + 1317
49  QtGui                         	       0x111003c8d QAction::activate(QAction::ActionEvent) + 395
50  QtCore                        	       0x111deba49 QObject::event(QEvent*) + 275
51  QtWidgets                     	       0x10f1e9489 QApplicationPrivate::notify_helper(QObject*, QEvent*) + 247
52  QtWidgets                     	       0x10f1ea256 QApplication::notify(QObject*, QEvent*) + 532
53  QtCore                        	       0x111db6a09 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 157
54  QtCore                        	       0x111db74a3 QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 387
55  libqcocoa.dylib               	       0x10e2a8511 QCocoaEventDispatcherPrivate::processPostedEvents() + 193
56  libqcocoa.dylib               	       0x10e2a8d03 QCocoaEventDispatcherPrivate::postedEventsSourceCallback(void*) + 423
57  CoreFoundation                	    0x7ff80f18d7d6 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
58  CoreFoundation                	    0x7ff80f18d779 __CFRunLoopDoSource0 + 157
59  CoreFoundation                	    0x7ff80f18d548 __CFRunLoopDoSources0 + 215
60  CoreFoundation                	    0x7ff80f18c1b8 __CFRunLoopRun + 919
61  CoreFoundation                	    0x7ff80f18b859 CFRunLoopRunSpecific + 557
62  HIToolbox                     	    0x7ff81a07ea09 RunCurrentEventLoopInMode + 292
63  HIToolbox                     	    0x7ff81a07e646 ReceiveNextEventCommon + 201
64  HIToolbox                     	    0x7ff81a07e561 _BlockUntilNextEventMatchingListInModeWithFilter + 66
65  AppKit                        	    0x7ff81275dc61 _DPSNextEvent + 880
66  AppKit                        	    0x7ff813071dc0 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1273
67  AppKit                        	    0x7ff81274f075 -[NSApplication run] + 603
68  libqcocoa.dylib               	       0x10e2a7afb QCocoaEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 1951
69  QtCore                        	       0x111dbd508 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 362
70  QtCore                        	       0x111db6e25 QCoreApplication::exec() + 119
71  cockatrice                    	       0x104b5ed57 main + 3975
72  dyld                          	       0x205685366 start + 1942
```

_______________________________________________________________________________________

**Steps to reproduce:**
 - Go to ```Cockatrice``` > ```Preferences``` > ```Sound```
 - Click on the ```Enable sounds```
 - The program crash :'(
 
 Thanks for your work on this great project, and thanks by advance for your help :)

**Comment by ebbit1q** [comment]
hey, can you try if using the latest beta you still experience this issue? https://github.com/Cockatrice/Cockatrice/releases

**Comment by alex-mehdi-dev** [comment]
> hey, can you try if using the latest beta you still experience this issue? [Cockatrice/Cockatrice/releases](https://github.com/Cockatrice/Cockatrice/releases)

Hey,

I try it but impossible to install it 
```“cockatrice” is damaged and can’t be opened. You should move it to the Trash.```

Does something is missing during the build?

**Comment by alex-mehdi-dev** [comment]
> that's strange, which download did you use? could you try downloading the Cockatrice-2.9.1-beta.2-macOS-14_Sonoma.dmg one?

Yes I try with this image twice.

**Comment by alex-mehdi-dev** [comment]
Hi,

Yes I try everything I can (except building it locally which I had done on a Raspberry PI a long time a go which works perfectly).

To investigate in more detail, I also try for the beta one. Each time, I erase the old version and reinstall ```cockatrice``` and ```oracle``` from the launcher on the same operating system ```macOS 14.5```. I have done it again this afternoon (no operating system update since the creation of this ticket).

<img width="993" alt="image" src="https://github.com/user-attachments/assets/bb5d125c-ec3e-46a1-b07c-b5a6f73a2998">

Here are the results:
```Cockatrice-Rings of the Wild-2.9.0-macOS-11_Big_Sur.dmg``` -> ✅
```Cockatrice-2.9.1-beta-macOS-14_Sonoma.dmg```                     -> ❌
```Cockatrice-2.9.1-beta.2-macOS-14_Sonoma.dmg```                  -> ❌

It seems that the packaging with Sonoma versions fail :S

**Comment by ebbit1q** [comment]
ah, thank you for your feedback, it is likely an issue with the version of the qt library we use for the sonoma images then

**Comment by tooomm** [comment]
Did you try the non-Somona dmg as well?
That should run fine on newer OS versions.

**Comment by alex-mehdi-dev** [comment]
> Did you try the non-Sonoma dmg as well? That should run fine on newer OS versions.

Yes, I try.
```Cockatrice-Rings of the Wild-2.9.0-macOS-11_Big_Sur.dmg``` can be installed ✅, but it crash when enable sound.

**Comment by tooomm** [comment]
I was referring to

Cockatrice-2.9.1-beta.2-macOS-12_Monterey_and_13_Ventura.dmg

**Comment by alex-mehdi-dev** [comment]
Hi,

I try the `Cockatrice-2.9.1-beta.2-macOS-12_Monterey_and_13_Ventura.dmg` image. 
✅ it can be installed and started.
✅ enabling sound does not make it crash.
❌  system sound engine seems not to works (clicking on the testing button does not emit sound, and sound are not emit during a game).
❌ at the first launch I got a crash. But then no crash. I guess this is due to permissions. Indeed, when we start the application on macOs, we need to explicitly open with a right click > `Open` in the `Application` directory in order to confirm the start of not verified application. And because I did not do it first on `oracle` to allow the application, starting it with Cockatrice crash because it does not have the permission. Thus, not need to worry about this. For me it is not a bug, it is the normal process to use a not verified packaged application.

Here is the report : 

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               cockatrice [5304]
Path:                  /Applications/cockatrice.app/Contents/MacOS/cockatrice
Identifier:            com.cockatrice.Cockatrice
Version:               2.9.1 (2.9.1)
Code Type:             X86-64 (Translated)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2024-08-18 11:39:40.3767 +0200
OS Version:            macOS 14.6.1 (23G93)
Report Version:        12
Anonymous UUID:        AC36FBA9-EC66-179B-E581-4B821027D1E6


Time Awake Since Boot: 3300 seconds

System Integrity Protection: enabled

Notes:
PC register does not match crashing frame (0x0 vs 0x1108DACB4)

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000008
Exception Codes:       0x0000000000000001, 0x0000000000000008

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [5304]

VM Region Info: 0x8 is not in any region.  Bytes before following region: 4342128632
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      __TEXT                      102cfa000-1033ba000    [ 6912K] r-x/r-x SM=COW  /Applications/cockatrice.app/Contents/MacOS/cockatrice

Error Formulating Crash Report:
PC register does not match crashing frame (0x0 vs 0x1108DACB4)

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   QtCore                        	       0x1108dacb4 QObject::deleteLater() + 26
1   cockatrice                    	       0x102f39f1a MainWindow::exitCardDatabaseUpdate() + 26
2   QtCore                        	       0x1108df843 void doActivate<false>(QObject*, int, void**) + 1365
3   QtCore                        	       0x110a201f8 QProcessPrivate::processFinished() + 180
4   QtCore                        	       0x1108df843 void doActivate<false>(QObject*, int, void**) + 1365
5   QtCore                        	       0x1108ed6b0 QSocketNotifier::event(QEvent*) + 268
6   QtWidgets                     	       0x10d47bfab QApplicationPrivate::notify_helper(QObject*, QEvent*) + 247
7   QtWidgets                     	       0x10d47cd31 QApplication::notify(QObject*, QEvent*) + 541
8   QtCore                        	       0x1108a4dbb QCoreApplication::notifyInternal2(QObject*, QEvent*) + 157
9   QtCore                        	       0x1109c7586 qt_mac_socket_callback(__CFSocket*, unsigned long, __CFData const*, void const*, void*) + 166
10  CoreFoundation                	    0x7ff80617330e __CFSocketPerformV0 + 950
11  CoreFoundation                	    0x7ff8061497d6 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
12  CoreFoundation                	    0x7ff806149779 __CFRunLoopDoSource0 + 157
13  CoreFoundation                	    0x7ff8061495ac __CFRunLoopDoSources0 + 315
14  CoreFoundation                	    0x7ff8061481b8 __CFRunLoopRun + 919
15  CoreFoundation                	    0x7ff806147859 CFRunLoopRunSpecific + 557
16  HIToolbox                     	    0x7ff811040a09 RunCurrentEventLoopInMode + 292
17  HIToolbox                     	    0x7ff811040816 ReceiveNextEventCommon + 665
18  HIToolbox                     	    0x7ff811040561 _BlockUntilNextEventMatchingListInModeWithFilter + 66
19  AppKit                        	    0x7ff809719171 _DPSNextEvent + 880
20  AppKit                        	    0x7ff80a02daf0 -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1273
21  AppKit                        	    0x7ff80970a585 -[NSApplication run] + 603
22  libqcocoa.dylib               	       0x10c4a49c3 QCocoaEventDispatcher::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) + 2409
23  QtCore                        	       0x1108ab79d QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 367
24  QtCore                        	       0x1108a520b QCoreApplication::exec() + 119
25  cockatrice                    	       0x102e29815 main + 3637
26  dyld                          	       0x2036c7345 start + 1909
```

TLDR: 
- the image does not crash which is a good new for next releases on macOS if we package this application this way.
- enabling sound does not work, but this is a minor problem.

If you want to close the issue, it is up to you (the application does not crash anymore). Thanks again for you amazing work 😁

**Comment by tooomm** [comment]
>If you want to close the issue, it is up to you (the application does not crash anymore). Thanks again for you amazing work 😁

Not so sure if closing is the right approach here.
Users can still pick the wrong package and crash their client.
Which Apple device are you using, is it an Apple/ARM/M CPU or Intel/x86?

Sound would be good to have working...
Maybe some macOS guys are having ideas.

**Comment by alex-mehdi-dev** [comment]
I'm on the last macbook pro m3 Apple.

**Comment by alex-mehdi-dev** [comment]
Ok here something interesting. Recently I got an update on Mac. I used to play with Cockatrice-2.9.1-beta-macOS-14_Sonoma.dmg. I was wondering if I can install back the last stable version `Cockatrice-Rings of the Wild-2.9.0-macOS-11_Big_Sur.dmg` for more stability. To my surprise, it will not work anymore :/ Thus I try other versions:

Cockatrice-Rings of the Wild-2.9.0-macOS-11_Big_Sur.dmg -> ❌ Crash on start.
Cockatrice-2.9.1-beta-macOS-14_Sonoma.dmg -> ❌ Crash on start.
Cockatrice-2.9.1-beta.2-macOS-14_Sonoma.dmg -> ❌ Crash on start.

In short, previous version does not work anymore. I try to install [Cockatrice-Rings of the Wild-2.9.0-macOS-11_Big_Sur.dmg](https://github.com/Cockatrice/Cockatrice/releases/download/2023-09-14-Release-2.9.0/Cockatrice-Rings.of.the.Wild-2.9.0-macOS-11_Big_Sur.dmg) in a desperate move. This version works perfectly, and the sound works too ! I don't how this version is packaged with Qt. But I guess, this is the right way. I'm not sure if you can really do something with this external unstability. But I want you let you know what I found.

**Comment by tooomm** [comment]
Thanks for providing all this input.

I'll add our config in CI for building the 2.9.0 release for macOS 10.15:
https://github.com/Cockatrice/Cockatrice/blob/2023-09-14-Release-2.9.0/.github/workflows/desktop-build.yml#L209-L216

**Comment by tooomm** [comment]
Hi @alex-mehdi-dev, as GitHub keeps updating the provided images for their GH Actions runner, and they removed macos-11, macos-12 and macos-13 (next month)...
The lowest we can target is macos-14 image with an older Xcode for macOS 13 builds.

Can you please test those new builds again (Intel & Apple M), and put the first lines from the debug log here as well for each?
https://github.com/Cockatrice/Cockatrice/actions/runs/11539857625

You are still on MacBook Pro M3 with macOS 14?

**Comment by alex-mehdi-dev** [comment]
Hi @tooomm 

I test the new build for [macOS13-dmg](https://github.com/Cockatrice/Cockatrice/actions/runs/11539857625/artifacts/2109008830) because I'm still on MacBook Pro M3 and macOS 14. The build seems to be damaged... I try to start it in command line to have more trace but nothing seems to be display in the console.

**Comment by tooomm** [comment]
I enabled the debug artifacts, you can get the build here: @alex-mehdi-dev
https://github.com/Cockatrice/Cockatrice/actions/runs/11634868549?pr=5040#artifacts

Please don't hesitate to try the Intel build as well, ARM-based Apple M Macs can emulate and run x86-based applications as well.

Thanks for helping to track this down!

**Comment by alex-mehdi-dev** [comment]
Hi,

I have tested both version:

`Cockatrice-2.10.0-custom(a78b744)-macOS13_Apple.dmg` -> ❌ Crash on start and no debug log available even if started in command line.

`Cockatrice-2.10.0-custom(a78b744)-macOS13_Intel.dmg` -> 
 ✅ Can be started.
 ❌ Sound does not work but the application does not crash when switching the option in the preferences.
Here is the log I can 
```
Client Version: 2.10.0-custom(a78b744) (2024-11-01)
Client Operating System: macOS 14.6
Build Architecture: 64-bit
Qt Version: 6.7.2
System Locale: en_FR
Install Mode: Standard
---------------------------------------------------------------------------
Theme changed: "Mine"
SoundEngine: enabling sound with 0 sounds
No QtMultimedia backends found. Only QMediaDevices, QAudioDevice, QSoundEffect, QAudioSink, and QAudioSource are available.
Failed to initialize QMediaPlayer "Not available"
Sound theme changed: "Default"
SoundEngine: enabling sound with 7 sounds
Unable to load qt translation "qt_" at "/Applications/cockatrice.app/Contents/Resources/translations"
Unable to load "cockatrice" translation "cockatrice_" at "/Applications/cockatrice.app/Contents/MacOS/../Resources/translations"
main(): starting main program
main(): MainWindow constructor finished
Spoilers Disabled
main(): ui.show() finished
PictureLoader: cache fail for "_trice_card_back_295401"
Startup: found config with current version
CardDatabase::loadCardDatabases start
[CockatriceXml4Parser] Trying to parse:  "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/cards.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/cards.xml" Status = 0 Cards = 31019 Sets = 779 "973ms"
[CockatriceXml4Parser] Trying to parse:  "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/tokens.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/tokens.xml" Status = 0 Cards = 31901 Sets = 779 "23ms"
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/spoiler.xml" Status = 4 Cards = 31901 Sets = 779 "0ms"
Loading Custom Set 0 ( "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/customsets/TK.xml" )
[CockatriceXml4Parser] Trying to parse:  "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/customsets/TK.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/aza/Library/Application Support/Cockatrice/Cockatrice/customsets/TK.xml" Status = 0 Cards = 31901 Sets = 779 "0ms"
CardDatabase::loadCardDatabases success
QObject::connect: No such slot UserInterfaceSettingsPage::setNotificationEnabled(Qt::CheckState)
Error: Spoiler.xml not found or not deleted
Card Database load status:  0
```

**Comment by alex-mehdi-dev** [comment]
Thus:
- ✅ `Cockatrice-2.10.0-custom(a78b744)-macOS13_Intel.dmg` seems to solve a problem. Indeed, with the last update of Mac, `Cockatrice-Rings of the Wild-2.9.0-macOS-11_Big_Sur.dmg` crash on start (it was not the case before the update). But the  `Cockatrice-Rings.of.the.Wild-2.9.0-macOS-10.15_Catalina.dmg` works. So this is a great news to at least having a new version working.

- ❌  `Cockatrice-2.10.0-custom(a78b744)-macOS13_Intel.dmg` still have a problem of sound (but this is not big deal).
- ❌  `Cockatrice-2.10.0-custom(a78b744)-macOS13_Intel.dmg` or 2.10.0 versions in general may have a problem with games filters. Maybe someone else can test this to check if I'm not wrong (I try to double test everything).

We are getting closer which a good news :)

Thanks again for your work !

**Comment by alex-mehdi-dev** [comment]
It seems that the filters works fine after all. Maybe it was a server problem or a bad usage on my own.

**Comment by ebbit1q** [comment]
the mystery of the sound not working remains, it's most likely an upstream qt bug though....

**Comment by tooomm** [comment]
There is a new sound fix in the most recent beta in case you want to try.

2.10.0-beta.22

**Comment by tooomm** [comment]
Do you have the same issues still with the new v2.10.0? @alex-mehdi-dev

**Comment by alex-mehdi-dev** [comment]
Hi @tooomm,

Sorry for the delay. I just install the following version [Cockatrice-Omenpath-2.10.2-macOS15_Apple.dmg](https://github.com/Cockatrice/Cockatrice/releases/download/2025-04-03-Release-2.10.2/Cockatrice-Omenpath-2.10.2-macOS15_Apple.dmg). It works perfectly 👍 The sound work, and I had no crash. I use the installer and it works perfectly too. Great work ! Maybe we can close my issue ?

**Comment by tooomm** [comment]
I assume you are running macOS 15 in the meantime then? :)

Out of curiosity, does the macOS 13 release (v2.10.2), build for Intel architecture, works flawlessly as well on your newer machine?

**Comment by tooomm** [comment]
Glad to hear (and thanks for checking the other package as well)!

Credit goes to all team and some very active new contributors! 🚀
